components/post_view/post_message_view/post_message_view.jsx

```diff
@@ -208,32 +236,55 @@ export default class PostMessageView extends React.PureComponent {
         const mentionKeys = [...this.props.mentionKeys, this.props.currentUser.username];
 
         const options = Object.assign({}, this.props.options, {
-            emojis: this.props.emojis,
-            siteURL: this.props.siteUrl,
+            emojis,
+            siteURL: siteUrl,
             mentionKeys,
             atMentions: true,
             channelNamesMap: getChannelsNameMapInCurrentTeam(store.getState()),
-            team: this.props.team
+            team
         });
 
-        const renderedSystemMessage = renderSystemMessage(this.props.post, options);
+        const renderedSystemMessage = renderSystemMessage(post, options);
         if (renderedSystemMessage) {
             return <div>{renderedSystemMessage}</div>;
         }
 
         let postId = null;
-        if (this.props.lastPostCount >= 0) {
-            postId = Utils.createSafeId('lastPostMessageText' + this.props.lastPostCount);
+        if (lastPostCount >= 0) {
+            postId = Utils.createSafeId('lastPostMessageText' + lastPostCount);
         }
 
-        let message = this.props.post.message;
-        const isEphemeral = Utils.isPostEphemeral(this.props.post);
-        if (this.props.compactDisplay && isEphemeral) {
+        let message = post.message;
+        const isEphemeral = Utils.isPostEphemeral(post);
+        if (compactDisplay && isEphemeral) {
             const visibleMessage = Utils.localizeMessage('post_info.message.visible.compact', ' (Only visible to you)');
             message = message.concat(visibleMessage);
         }
-        const htmlFormattedText = TextFormatting.formatText(message, options);
-        const postMessageComponent = this.postMessageHtmlToComponent(htmlFormattedText);
+
+        const isSystemAdmin = UserStore.isSystemAdminForCurrentUser();
```

Comment:
Can we move this logic into `renderSystemMessage` since that's where all the similar logic for rendering system messages is?

It would also be good if we could re-order this logic to something like
```
if (post.props && post.props.add_channel_member) {
    const isTeamAdmin = ...;
    const isChannelAdmin = ...;
    const isUserCanManageMembers = ...;

    ...
}
```
so that it's very clear that all of that logic is only used for that specific type of post.

Reply:
I agree.